pkg/southbound/synchronizer/synchronizer.go

```diff
@@ -210,25 +222,51 @@ func (sync Synchronizer) syncOperationalState(errChan chan<- events.DeviceRespon
 	if len(notifications) != 0 {
 		for _, notification := range notifications {
 			for _, update := range notification.Update {
-				pathStr := utils.StrPath(update.Path)
-				val := utils.StrVal(update.Val)
-				sync.operationalCache[pathStr] = val
-				subscribePaths = append(subscribePaths, utils.SplitPath(pathStr))
-			}
 
+				jsonVal := update.Val.GetJsonVal()
+				//strVla := utils.StrVal(update.Val)
+				//log.Info(strVla)
+				configValuesUnparsed, err := store.DecomposeTree(jsonVal)
+				if err != nil {
+					log.Error("Can't translate from json to values, skipping to next update", err)
+					break
+				}
+				configValues, err := jsonvalues.CorrectJSONPaths(configValuesUnparsed, sync.modelReadOnlyPaths)
+				if err != nil {
+					log.Error("Can't translate from config values to typed values, skipping to next update", err)
+					break
+				}
+				pathsAndValues := make(map[string]*change.TypedValue)
+				for _, p := range configValues {
+					pathsAndValues[p.Path] = &p.TypedValue
+				}
+				sync.operationalCache[sync.ID] = pathsAndValues
+
+			}
 		}
-	} else if sync.modelReadOnlyPaths != nil {
+	}
+	if sync.modelReadOnlyPaths != nil {
 		for _, path := range modelregistry.Paths(sync.modelReadOnlyPaths) {
 			subscribePaths = append(subscribePaths, utils.SplitPath(path))
 		}
+	} else {
+		errMp := fmt.Errorf("No model plugin, cant work in operational state cache")
+		log.Error(errMp)
+		errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorMissingModelPlugin,
+			sync.key.DeviceID, errMp)
+		return
 	}
 
 	if len(subscribePaths) == 0 {
```

Comment:
same again - if there are no subscribe paths then should not be going the get

Reply:
done